meilisearch-http/src/backup.rs

```diff
@@ -0,0 +1,369 @@
+use tempfile::TempDir;
+
+#[derive(Debug, Serialize, Deserialize, Copy, Clone)]
+enum BackupVersion {
+    V1,
+}
+
+impl BackupVersion {
+    fn current() -> Self { Self::V1 }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BackupMetadata {
+    indexes: Vec<crate::routes::index::IndexResponse>,
+    db_version: String,
+    backup_version: BackupVersion,
+}
+
+impl BackupMetadata {
+    pub fn new(indexes: Vec<crate::routes::index::IndexResponse>, db_version: String) -> Self {
+        BackupMetadata {
+            indexes: indexes,
+            db_version: db_version,
+            backup_version: BackupVersion::current(),
+        }
+    }
+
+    fn from_path(folder_path: &Path) -> Result<Self, Error> {
+        let path = folder_path.join("metadata.json");
+        let file = File::open(path)?;
+        let reader = std::io::BufReader::new(file);
+        let metadata = serde_json::from_reader(reader)?;
+
+        Ok(metadata)
+    }
+
+    fn to_path(&self, folder_path: &Path) -> Result<(), Error> {
+        let path = folder_path.join("metadata.json");
+        let file = File::create(path)?;
+
+        serde_json::to_writer(file, &self)?;
+
+        Ok(())
+    }
+}
+
+fn settings_from_path(folder_path: &Path) -> Result<Settings, Error> {
+    let path = folder_path.join("settings.json");
+    let file = File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let metadata = serde_json::from_reader(reader)?;
+
+    Ok(metadata)
+}
+
+fn settings_to_path(settings: &Settings, folder_path: &Path) -> Result<(), Error> {
+    let path = folder_path.join("settings.json");
+    let file = File::create(path)?;
+
+    serde_json::to_writer(file, settings)?;
+
+    Ok(())
+}
+
+fn import_index_v1(
+    data: &Data,
+    backup_path: &Path,
+    index_uid: &str,
+    document_batch_size: usize,
+    write_txn: &mut RwTxn<MainT>,
+) -> Result<(), Error> {
+    let index = data
+        .db
+        .open_index(index_uid)
+        .ok_or(Error::index_not_found(index_uid))?;
+
+    let index_path = &backup_path.join(index_uid);
+
+    let settings = settings_from_path(&index_path)?;
+    let settings = settings.to_update().or_else(|_e| Err(Error::backup_failed()))?;
+    apply_settings_update(write_txn, &index, settings)?;
+
+    let documents = {
+        let file = File::open(&index_path.join("documents.jsonl"))?;
+        let reader = std::io::BufReader::new(file);
+        let deserializer = serde_json::Deserializer::from_reader(reader);
+        deserializer.into_iter::<serde_json::Value>()
+    };
+
+    let mut values = Vec::with_capacity(document_batch_size);
+    for document in documents {
+        let vec = serde_json::to_vec(&document?)?;
+        let document = serde_json::from_slice(&vec)?;
+        values.push(document);
+        if values.len() == document_batch_size { 
+            apply_documents_addition(write_txn, &index, values.clone())?;
+            values.clear();
```

Comment:
```suggestion
            let batch = mem::replace(&mut value, Vec::with_capacity(document_batch_size));
            apply_documents_addition(write_txn, &index, batch)?;
```

https://doc.rust-lang.org/std/mem/fn.replace.html